src/Sylius/Bundle/ResourceBundle/Controller/ResourceController.php

```diff
@@ -316,11 +328,18 @@ public function updateAction(Request $request)
                 return $this->redirectHandler->redirectToResource($configuration, $resource);
             }
 
-            if ($configuration->hasStateMachine()) {
-                $this->stateMachine->apply($configuration, $resource);
+            try {
+                $this->resourceUpdater->applyTransitionAndFlush($resource, $configuration, $this->manager);
+            } catch (ResourceException $exception) {
+                if (!$configuration->isHtmlRequest()) {
+                    return $this->viewHandler->handle($configuration, View::create($form, Response::HTTP_CONFLICT));
```

Comment:
How about [this idea](https://github.com/Sylius/Sylius/pull/7425#discussion_r101492048) with the response code?